tests/unit/integrations_kafka_consumer.cpp

```diff
@@ -496,8 +484,68 @@ TEST_F(ConsumerTest, ConsumerStatus) {
   check_info(consumer.Info());
   consumer.Start();
   check_info(consumer.Info());
-  consumer.StartIfStopped();
-  check_info(consumer.Info());
   consumer.StopIfRunning();
   check_info(consumer.Info());
 }
+
+TEST_F(ConsumerTest, LimitBatches_CannotStartIfAlreadyRunning) {
+  static constexpr auto kLimitBatches = 3;
+
+  auto info = CreateDefaultConsumerInfo();
+
+  auto consumer_function = [](const std::vector<Message> &messages) mutable {};
+
+  auto consumer = CreateConsumer(std::move(info), std::move(consumer_function));
+
+  consumer->Start();
+  ASSERT_TRUE(consumer->IsRunning());
+
+  EXPECT_THROW(consumer->StartWithLimit(kLimitBatches), ConsumerRunningException);
+
+  EXPECT_TRUE(consumer->IsRunning());
+
+  consumer->Stop();
+  EXPECT_FALSE(consumer->IsRunning());
+}
+
+TEST_F(ConsumerTest, LimitBatches_SendingMoreThanLimit) {
+  /*
+  We send more messages than the BatchSize*LimitBatches:
+  -Consumer should receive 2*3=6 messages.
+  -Consumer should not be running afterwards.
+  */
+  static constexpr auto kBatchSize = 2;
+  static constexpr auto kLimitBatches = 3;
+  static constexpr auto kNOfMessagesToSend = 20;
+  static constexpr auto kNOfMessagesExpected = kBatchSize * kLimitBatches;
+  static constexpr auto kBatchInterval =
+      std::chrono::seconds{2};  // We do not want the batch interval to be the limiting factor here.
+
+  auto info = CreateDefaultConsumerInfo();
+  info.batch_size = kBatchSize;
+  info.batch_interval = kBatchInterval;
+
+  static constexpr std::string_view kMessage = "LimitBatchesTestMessage";
+
+  auto expected_messages_received = true;
+  auto nOfMessagesReceived = 0;
+  auto consumer_function = [&expected_messages_received,
+                            &nOfMessagesReceived](const std::vector<Message> &messages) mutable {
+    nOfMessagesReceived += messages.size();
+    for (const auto &message : messages) {
+      expected_messages_received &= (kMessage == std::string_view(message.Payload().data(), message.Payload().size()));
+    }
+  };
+
+  auto consumer = CreateConsumer(std::move(info), std::move(consumer_function));
+
+  for (auto sent_messages = 0; sent_messages <= kNOfMessagesToSend; ++sent_messages) {
+    cluster.SeedTopic(kTopicName, kMessage);
+  }
+
+  consumer->StartWithLimit(kLimitBatches);
+
+  EXPECT_FALSE(consumer->IsRunning());
+  EXPECT_EQ(nOfMessagesReceived, kNOfMessagesExpected);
+  EXPECT_TRUE(expected_messages_received) << "Some unexpected message have been received";
+}
```

Comment:
I think you pre-commit hook is not set up properly or you are not on the latest master. It should fix this missing new line.